packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -25,15 +30,21 @@ let portalObserver: MutationObserver | undefined;
 
 const portalObserverConfig: MutationObserverInit = {
     childList: true,
-    subtree: true,
+    subtree: false,
 };
 
 function adoptChildNode(node: Node, fn: ShadowRootResolver, shadowToken: string | undefined) {
-    if (getShadowRootResolver(node) === fn) {
+    const previousNodeShadowResolver = getShadowRootResolver(node);
+    if (previousNodeShadowResolver === fn) {
         return; // nothing to do here, it is already correctly patched
     }
     setShadowRootResolver(node, fn);
-    if (node instanceof Element) {
+    // Root LWC elements can't contain slots, therefore we stop listening mutations.
+    if (node instanceof Element && !isHostElement(node)) {
+        if (isUndefined(previousNodeShadowResolver)) {
+            // we only care about Element without shadowResolver (no MO.observe has been called)
+            MutationObserverObserve.call(portalObserver, node, portalObserverConfig);
+        }
         setShadowToken(node, shadowToken);
```

Comment:
@caridy why do we use this? This code path is not adding it to the host elements.

Reply:
this has to be accommodated, it should be added to custom elements, but not to the children elements of a it. I think we can keep the old condition `if (node instanceof Element) {`, then do the `setShadowtoken()`, then checking if it is not host to do the rest, which is install mutation and attempt to recursively patch all children.